proposals/NNNN-resource-attributes-in-dxil-and-spirv.md

```diff
@@ -0,0 +1,386 @@
+| Texture3D                          | SRV  | vec4   | -   | 3D   | -   | -        | -     | -   | -   |
+| TextureCUBE                        | SRV  | vec4   | -   | Cube | -   | -        | -     | -   | -   |
+| TextureCUBEArray                   | SRV  | vec4   | -   | Cube | -   | -        | yes   | -   | -   |
+| RWTexture1D                        | UAV  | vec4   | -   | 1D   | -   | -        | -     | -   | -   |
+| RWTexture1DArray                   | UAV  | vec4   | -   | 1D   | -   | -        | yes   | -   | -   |
+| RWTexture2D                        | UAV  | vec4   | -   | 2D   | -   | -        | -     | -   | -   |
+| RWTexture2DArray                   | UAV  | vec4   | -   | 2D   | -   | -        | yes   | -   | -   |
+| RWTexture2DMS                      | UAV  | vec4   | -   | 2D   | yes | -        | -     | -   | -   |
+| RWTexture2DMSArray                 | UAV  | vec4   | -   | 2D   | yes | -        | yes   | -   | -   |
+| RWTexture3D                        | UAV  | vec4   | -   | 3D   | -   | -        | -     | -   | -   |
+| RasterizerOrderedTexture1D         | UAV  | vec4   | yes | 1D   | -   | -        | -     | -   | -   |
+| RasterizerOrderedTexture1DArray    | UAV  | vec4   | yes | 1D   | -   | -        | yes   | -   | -   |
+| RasterizerOrderedTexture2D         | UAV  | vec4   | yes | 2D   | -   | -        | -     | -   | -   |
+| RasterizerOrderedTexture2DArray    | UAV  | vec4   | yes | 2D   | -   | -        | yes   | -   | -   |
+| RasterizerOrderedTexture3D         | UAV  | vec4   | yes | 3D   | -   | -        | -     | -   | -   |
+| FeedbackTexture2D                  | SRV  | -      | -   | -    | -   | fbtype   | -     | -   | -   |
+| FeedbackTexture2DArray             | SRV  | -      | -   | -    | -   | fbtype   | yes   | -   | -   |
+| Buffer                             | SRV  | vec4   | -   | -    | -   | -        | -     | -   | -   |
+| RWBuffer                           | UAV  | vec4   | -   | -    | -   | -        | -     | -   | -   |
+| RasterizerOrderedBuffer            | UAV  | vec4   | yes | -    | -   | -        | -     | -   | -   |
+| ByteAddressBuffer                  | SRV  | -      | -   | -    | -   | -        | -     | yes | -   |
+| RWByteAddressBuffer                | UAV  | -      | -   | -    | -   | -        | -     | yes | -   |
+| RasterizerOrderedByteAddressBuffer | UAV  | -      | yes | -    | -   | -        | -     | yes | -   |
+| StructuredBuffer                   | SRV  | struct | -   | -    | -   | -        | -     | yes | -   |
+| RWStructuredBuffer                 | UAV  | struct | -   | -    | -   | -        | -     | yes | -   |
+| RasterizerOrderedStructuredBuffer  | UAV  | struct | yes | -    | -   | -        | -     | yes | -   |
+| AppendStructuredBuffer             | UAV  | struct | -   | -    | -   | -        | -     | yes | -   |
+| ConsumeStructuredBuffer            | UAV  | struct | -   | -    | -   | -        | -     | yes | -   |
```

Comment:
To generate or use these types, the SPIR-V backend will have to know the layout of any structs. In DXC, we have a few different layout options. I feel like it should be the FE's job to specify the layout. Let me know what you think.